core/connection/manager.go

```diff
@@ -132,24 +132,46 @@ func (manager *connectionManager) Connect(consumerID identity.Identity, proposal
 
 	connection, err := manager.newConnection(proposal.ServiceType, stateChannel, statisticsChannel)
 	if err != nil {
+		manager.sendDialogStatus(dialog, communication.StatusConnectionFailed, err.Error())
 		return err
 	}
 
 	sessionDTO, paymentInfo, err := manager.createSession(connection, dialog, consumerID, proposal)
 	if err != nil {
+		manager.sendDialogStatus(dialog, communication.StatusSessionFailed, err.Error())
 		return err
 	}
 
 	err = manager.launchPayments(paymentInfo, dialog, consumerID, providerID)
 	if err != nil {
+		manager.sendDialogStatus(dialog, communication.StatusPaymentsFailed, err.Error())
 		return err
 	}
 
 	err = manager.startConnection(connection, consumerID, proposal, params, sessionDTO, stateChannel, statisticsChannel)
-	if err == context.Canceled {
-		return ErrConnectionCancelled
+	if err != nil {
+		if err == context.Canceled {
+			return ErrConnectionCancelled
+		}
+		// TODO: Here we need to extract mode detailed error.
+		manager.sendDialogStatus(dialog, communication.StatusConnectionFailed, err.Error())
+		return err
+	}
+	manager.sendDialogStatus(dialog, communication.StatusConnectionOk, "")
```

Comment:
At this point, when dialog establishes we still do not know anything about actual consumer connectivity status (check for ipify ip change). This dialog status logic you implemented can go within dialog establishment, but for actual consumer connectivity status you need to produce a separate message, i.e. as @vkuznecovas done in pingpong package.